lapce-data/src/editor.rs

```diff
@@ -2408,3 +2405,43 @@ fn apply_code_action(
         log::error!("Failed to convert code action edit Position to offset");
     }
 }
+
+/// Checks if completion should be triggered if the received command
+/// is one that inserts whitespace or deletes whitespace
+fn show_completion(
+    cmd: &EditCommand,
+    doc: &str,
+    deltas: &[(RopeDelta, InvalLines)],
+) -> bool {
+    let show_completion = match cmd {
+        EditCommand::DeleteBackward | EditCommand::DeleteForward => {
+            let start = match &deltas[0].0.els[0] {
+                xi_rope::DeltaElement::Copy(_, start) => start,
+                _ => &0,
+            };
+
+            let end = match &deltas[0].0.els[1] {
+                xi_rope::DeltaElement::Copy(end, _) => end,
+                _ => &0,
+            };
+
+            if start > &0 && end > &0 && end > start {
+                let slice = &doc[*start..*end];
```

Comment:
You could use the ```slice_to_cow``` on the rope to make it more efficient without converting the rope to a string.

https://github.com/lapce/lapce/blob/master/lapce-core/src/buffer.rs#L931

Reply:
Would you pass the `Rope` to `show_completion(...)`? Because `slice_to_cow` takes a `Range` as argument it would end up something like this:
```rust
fn show_completion(cmd: &EditCommand, doc: &Rope, deltas: &[(RopeDelta, InvalLines)]) -> bool {
    [...]
    if start > &0 && end > start {
        !doc.slice_to_cow(*start..*end)
            .chars()
            .all(|c| c.is_whitespace() || c.is_ascii_whitespace())
    } else {
        true
}
```
I'll have to try if this works and update with results 😄